application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarkPaginatorInteractor.java

```diff
@@ -0,0 +1,479 @@
+                    currentPageIndex, bookmarksToRemoveChannelIDs);
+
+            return generateComponentId(args);
+        };
+
+        List<SelectOption> selectMenuRemoveOptions = pageEntries.stream().map(pageEntry -> {
+            ThreadChannel channel = jda.getThreadChannelById(pageEntry.bookmark.getChannelId());
+            String channelIDString = String.valueOf(pageEntry.bookmark.getChannelId());
+            int bookmarkNumber = pageEntry.bookmarkNumber;
+
+            String label = channel != null ? "%d. %s".formatted(bookmarkNumber, channel.getName())
+                    : "Delete bookmark %d".formatted(bookmarkNumber);
+
+            return SelectOption.of(label, channelIDString);
+        }).toList();
+
+        String selectMenuRemoveId = generateRemoveComponentId.apply(SELECTMENU_REMOVE_NAME);
+        SelectMenu selectMenuRemove = SelectMenu.create(selectMenuRemoveId)
+            .setPlaceholder("Select bookmarks to delete")
+            .addOptions(selectMenuRemoveOptions)
+            .setDefaultValues(parseListToStringList(bookmarksToRemoveChannelIDs))
+            .setRequiredRange(0, selectMenuRemoveOptions.size())
+            .build();
+
+        String buttonRemoveId = generateRemoveComponentId.apply(BUTTON_REMOVE_NAME);
+        Button buttonRemove = Button.danger(buttonRemoveId, BUTTON_REMOVE_LABEL)
+            .withDisabled(bookmarksToRemoveChannelIDs.isEmpty());
+
+        return List.of(ActionRow.of(selectMenuRemove), ActionRow.of(buttonRemove));
+    }
+
+    /**
+     * Removed the selected bookmarks from the bookmarks list and the database.
+     *
+     * @param bookmarks The bookmarks list
+     * @param event The component interaction event
+     * @param args The remove component arguments
+     */
+    private void removeSelectedBookmarks(List<BookmarksRecord> bookmarks,
+            ComponentInteraction event, RemoveComponentArguments args) {
+        long userID = event.getUser().getIdLong();
+
+        Predicate<BookmarksRecord> removePredicate =
+                bookmark -> args.bookmarksToRemoveChannelIDs.contains(bookmark.getChannelId());
+
+        bookmarks.removeIf(removePredicate);
+        bookmarksSystem.removeBookmarks(userID, args.bookmarksToRemoveChannelIDs);
+    }
+
+    /**
+     * Calculates which bookmarks to display as page entries.
+     *
+     * @param bookmarks The bookmarks list
+     * @param pageIndex The page index the entries should be for
+     * @return A list of entries to be displayed on the requested page
+     */
+    private static List<PageEntry> getPageEntries(List<BookmarksRecord> bookmarks, int pageIndex) {
+        int indexStart = pageIndex * ENTRIES_PER_PAGE;
+        int indexEndMax = bookmarks.size();
+        int indexEnd = Math.min(indexStart + ENTRIES_PER_PAGE, indexEndMax);
+
+        return bookmarks.subList(indexStart, indexEnd)
+            .stream()
+            .map(b -> new PageEntry(bookmarks.indexOf(b) + 1, b))
```

Comment:
Rename b to bookmark
```suggestion
            .map(bookmark -> new PageEntry(bookmarks.indexOf(b) + 1, bookmark))
```